tests/infrastructure/instance_types/utils.py

```diff
@@ -13,3 +17,27 @@ def get_mismatch_vendor_label(resources_list):
 def assert_mismatch_vendor_label(resources_list):
     failed_labels = get_mismatch_vendor_label(resources_list=resources_list)
     assert not failed_labels, f"The following resources have miss match vendor label: {failed_labels}"
+
+
+def get_controller_revision(
+    vm_instance: VirtualMachine, ref_type: Literal["instancetype", "preference"]
+) -> ControllerRevision:
+    ref_mapping = {
+        "instancetype": vm_instance.instance.status.instancetypeRef.controllerRevisionRef.name,
+        "preference": vm_instance.instance.status.preferenceRef.controllerRevisionRef.name,
+    }
+
+    return ControllerRevision(
+        name=ref_mapping[ref_type],
+        namespace=vm_instance.namespace,
+    )
+
+
+def validate_revision_name_update(vm_for_test: VirtualMachine, old_revision_name: str, updated_memory: str):
+    guest_memory = vm_for_test.vmi.instance.spec.domain.memory.guest
+    assert vm_for_test.instance.spec.instancetype.revisionName != old_revision_name, (
+        "The revisionName is still {old_revision_name}, not updated after editing"
+    )
+    assert guest_memory == updated_memory, (
+        "The Guest Memory in VMI is {guest_memory}, not updated to {updated_memory} after editing"
+    )
```

Comment:
considering the function name, this assert should be a separate function..

Reply:
function renamed.